src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -568,8 +612,12 @@ private void uploadFleetStatusServiceData(Set<GreengrassService> greengrassServi
                 .messageType(MessageType.fromTrigger(trigger))
                 .deploymentInformation(deploymentInformation)
                 .build();
-
-        publisher.publish(fleetStatusDetails, components);
+        try {
+            Thread.sleep(10_000); // Prevent messages sent too frequently and cloud received them out of order
+            publisher.publish(fleetStatusDetails, components);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
why runtime exception?

Reply:
if you catch an interrupted exception, you should try to interrupt that thread using `Thread.currentThread().interrupt()`